### DISCUSSION TITLE
How to find Checksum Information

### DISCUSSION THREAD

**Discussion by bucanero** [bucanero/apollo-ps3#30] [General]
Please refer to the [How to find Checksum information](https://github.com/bucanero/apollo-ps3/wiki/How-to-find-Checksum-information) wiki article for the latest information.

**Comment by bucanero** [comment]
Tools:
- [ps3_game_integrity.zip](https://github.com/bucanero/apollo-ps3/files/5719668/ps3_game_integrity.zip)
- [Algorithm Tool.zip](https://github.com/bucanero/apollo-ps3/files/5719689/Algorithm.Tool.zip)

Hashes:
- http://www.hashemall.com/
- https://hash.online-convert.com/

**Reply by aldostools** [reply]
BSD includes the "BSD Patch Creator" that is a tool that I created to help to create BSD cheat scripts and to find the checksums.

To use it, the save must be decrypted (the file ~files_decrypted_by_pfdtool.txt must exists).
1-From Bruteforce SaveData, select the decrypted SAVE file on the green box.
2-Right-click on the file and select "Open Selected File on Patch Creator"
3-On the HEX editor panel, select the first byte where the checksum is found. It's usually found at the top or at the end.
4-Press Checksum button or press F7.
5-The Search Checksum dialog has 2 modes of finding:
a) Press F3 to test all the algorithms in the current offset
b) Press Search button to test various calculation ranges with all the supported algorithms).
Here you must provide the initial range, the end of range and the stop offset (for the initial range or the end range if Reverse Search is checked). Uncheck Step 4 if the increment of the search address must be 1.

The calculated checksum for the current range and it's reflected value are displayed in a box. 

Tip: Use the Up/Down buttons of the different parameters to search manually.

Once a matching checksum is found, a green box with a checkmark will be shown and a CODE button will appear.
Press the CODE button to view the script code and copy it to the clipboard.

**Reply by bucanero** [reply]
thanks @aldostools for the additional information! 👍 

I've just recovered another post from ps3cheating about [BSD Patch Creator](https://github.com/bucanero/apollo-ps3/wiki/How-to-find-cheats-(BSD-Patch-Creator))

I know that many sites/links are long gone but at least we can keep track of some of that information here for future reference 😃
